drivers/include/sdp3x.h

```diff
@@ -0,0 +1,199 @@
+/*
+ * Copyright (C) 2019 Dirk Ehmen
+ *               2020 Nishchay Agrawal
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    drivers_sdp3x SDP3x temperature and differential pressure sensor
+ * @ingroup     drivers_sensors
+ * @{
+ * @file
+ * @brief       Device driver interface for the SDP3x sensor.
+ *
+ * @author      Dirk Ehmen       <ehmen@ibr.cs.tu-bs.de>
+ * @author      Nishchay Agrawal <f2016088@pilani.bits-pilani.ac.in>
+ * @}
+ */
+
+#ifndef SDP3X_H
+#define SDP3X_H
+
+#include "saul.h"
+#include "xtimer.h"
+#include "mutex.h"
+#include "periph/i2c.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief   SDP3x models
+ */
+typedef enum {
+    SDP3x_31 = 1,
+    SDP3x_32 = 2
+} sdp3x_model_t;
+
+/**
+ * @brief   Device initialization parameters
+ */
+typedef struct {
+    i2c_t i2c_dev;                  /**< I2C device which is used */
+    uint8_t i2c_addr;               /**< I2C address */
+    sdp3x_model_t model;            /**< Model Number */
+    gpio_t irq_pin;                 /**< IRQ pin to check data ready */
+} sdp3x_params_t;
+
+/**
+ * @brief   Device descriptor for the SDP3x sensor
+ */
+typedef struct {
+    bool continuous_measurement;    /**< Device measuring data continuously */
+    sdp3x_params_t params;          /**< Device initialization parameters */
+    mutex_t mutex;                  /**< Mutex for this sdp3x dev */
+} sdp3x_t;
+
+/**
+ * @brief   Measurement from SDP3x sensor
+ */
+typedef struct {
+    int32_t differential_pressure;  /**< Differential Pressure measured*/
+    int32_t temperature;            /**< Temperature measured*/
+    int16_t dp_scale_factor;       /**< Scale factor for measurement*/
+} sdp3x_measurement_t;
+
+/**
+ * @brief   Status and error return codes
+ */
+enum {
+    SDP3x_OK = 0,             /**< all went as expected */
+    SDP3x_CRCERROR = -1,      /**< CRC-Check failed */
+    SDP3x_NODATA = -2,        /**< No Data available */
+    SDP3x_IOERROR = -3,       /**< I/O error */
+    SDP3x_WRONGSIZE = -4      /**< Wrong size of array for method */
+};
+
+/** Initialize SDP3x
+ *
+ *      @param dev         sdp3x device
+ *      @param params      sdp3x device params
+ *
+ *      @return            1 if device started
+ */
+int sdp3x_init(sdp3x_t *dev, const sdp3x_params_t *params);
+
+/** Start Continuous Measuring
+ *
+ *      @param dev          sdp3x device
+ *      @param tempComp     Temperature compensation:
+ *                          1 = Mass flow, 2 = Differential pressure
+ *      @param avg          Choose method for averaging values:
+ *                          0 = No averaging, 1 = Averaging till read
+ *
+ *      @return             1 if device started
+ */
+int8_t sdp3x_start_continuous(sdp3x_t *dev, uint8_t tempComp,
+                              uint8_t avg);
```

Comment:
```suggestion
int8_t sdp3x_start_continuous(sdp3x_t *dev, uint8_t flags);
```

I think the API would be better if you had flags instead of boolean arguments.

So the user would call `sdp3x_start_continuous(dev, SDP3X_FLAG_MASS_FLOW | SDP3X_FLAG_AVERAGE)` instead of `sdp3x_start_continuous(dev, 1, 1)` - much easier to understand what's going on.